### PR TITLE
DSND-1896: Upgrade structured logging version

### PR DESCRIPTION
**Pull request by johncook-ch**
* Added a `DataMapHolder` class to store a  thread local`DataMapHolder.Builder` instance that captures log data during message processing.
* Added an aspect, `MessageLoggingAspect`, that wraps around Kafka consumers to initialise the `DataMapHolder` with message data, then ensures it is clean when processing is complete.
* `MessageLoggingAspect` is applicable both `ChsDelta` and `ResourceChangedData`, however it will apply structured logging to consumers of all Kafka regardless of message payload type.
* Updated all logger calls to include the structured logging `DataMap` content.
* Refactored calls to `InternalApiClient` to set the request ID header with the one used on the initial call which will enable tracing calls from one service to another 
* Updated tests that assert log messages to include `DataMap` content.
* Fixed some linting issues that were breaking the build.

**Testing Notes:**
_Since possibly all paths through this service log messages, we need to ensure that we have not introduced errors where logging fails. This means that a general smoke test of all message types is needed._

[DSND-1896](https://companieshouse.atlassian.net/browse/DSND-1896)

[DSND-1862]: https://companieshouse.atlassian.net/browse/DSND-1862?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DSND-1896]: https://companieshouse.atlassian.net/browse/DSND-1896?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/company/links/service/AppointmentsListClient.java [hunk 1]
```diff
@@ -62,7 +64,8 @@ public OfficerList getAppointmentsList(String companyNumber) {
                 if ((ex.getHeaders().containsKey(CONTENT_LENGTH)
                         && ex.getHeaders().getContentLength() > 0)
                         || (ex.getContent() != null && ex.getContent().length() > 0)) {
-                    logger.error("Company-appointments service is not available");
+                    logger.error("Company-appointments service is not available",
+                            DataMapHolder.getLogMap());
                     throw new RetryableErrorException(
                             "Company-appointments service is not available", ex);
                 }
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company/links/service/AppointmentsListClient.java:71]: line 74 log is missing logmap

File: src/main/java/uk/gov/companieshouse/company/links/service/DeleteExemptionsClient.java [hunk 2]
```diff
@@ -39,28 +42,32 @@ public void patchLink(PatchLinkRequest linkRequest) {
...
+                        DataMapHolder.getLogMap());
                 throw new RetryableErrorException("Server error returned when processing "
                         + "delete exemptions link request", ex);
             } else if (ex.getStatusCode() == 409) {
                 logger.info("HTTP 409 Conflict returned; "
-                        + "company profile does not have an exemptions link already");
+                                + "company profile does not have an exemptions link already",
+                        DataMapHolder.getLogMap());
             } else if (ex.getStatusCode() == 404) {
                 logger.info("HTTP 404 Not Found returned; "
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company/links/service/DeleteExemptionsClient.java:54]: missing logmap

**Review by cgregoryCH — CHANGES_REQUESTED** [review]
Looks great, just a couple of missed log lines and a few unused imports

**Review by cgregoryCH — APPROVED** [review]

**Review by cgregoryCH — DISMISSED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Review by crowlandsCH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/company/links/logging/MessageLoggingAspect.java [hunk 3]
```diff
@@ -22,14 +24,13 @@ public MessageLoggingAspect(Logger logger) {
...
+    @Around("@annotation(org.springframework.kafka.annotation.KafkaListener)")
+    public Object manageStructuredLogging(ProceedingJoinPoint joinPoint) throws Throwable {
 
         try {
             Message<?> message = (Message<?>) joinPoint.getArgs()[0];
-            ResourceChangedData payload = (ResourceChangedData) message.getPayload();
-            String contextId = payload.getContextId();
-            DataMapHolder.initialise(contextId);
+            DataMapHolder.initialise(extractContextId(message.getPayload())
+                    .orElseGet(() -> "unknown"));
```
[the remark this answers is not in the document]
Comment (johncook-ch) [reply src/main/java/uk/gov/companieshouse/company/links/logging/MessageLoggingAspect.java]: Changed to return a random UUID string instead

**Review by cgregoryCH — APPROVED** [review]

**Review by crowlandsCH — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**